### PR TITLE
Fix typo in/update workflow

### PR DESCRIPTION
**Pull request by mattwthompson**
https://github.com/openforcefield/openff-toolkit/actions/runs/8311448282/job/22745135913#step:3:40

### CODE REVIEW

File: devtools/conda-envs/beta_rc_env.yaml [hunk 1]
```diff
@@ -14,19 +14,19 @@ dependencies:
...
   - openff-utilities >=0.1.5
-  - openff-interchange-base >=0.3.17
+  - openff-interchange-base >=0.3.23
   - openff-nagl-base ==0.3.0
   - openff-nagl-models ==0.1.0
     # Toolkit-specific
   - ambertools >=22
     # https://github.com/rdkit/rdkit/issues/7221
-  - rdkit >=2023.09.6
+  - rdkit <2023.09.3
```
[the remark this answers is not in the document]
Comment (j-wags) [reply devtools/conda-envs/beta_rc_env.yaml:29]: Thanks for adding the reference issue. I can't find any test failures due to this bug, am I looking in the right place?

[the remark this answers is not in the document]
Comment (mattwthompson) [reply devtools/conda-envs/beta_rc_env.yaml:29]: The builds that run mypy aren't pulling down the most recent RDKit version, let's see if I can trigger it ...

[the remark this answers is not in the document]
Comment (mattwthompson) [reply devtools/conda-envs/beta_rc_env.yaml:29]: Okay, here's one: https://github.com/openforcefield/openff-toolkit/actions/runs/8196534218/job/22444763324#step:15:21

That caused me to make this change (and report upstream): https://github.com/openforcefield/openff-toolkit/pull/1835

I stumbled across pretty much the same thing now (if forcing version 2023.09.6): 2023.09.6: https://github.com/openforcefield/openff-toolkit/actions/runs/8330217940/job/22794360609?pr=1839#step:14:21

I suspect it's a packaging change associated with the builds of version 2023.09.6 and not a code change, but I didn't look to deeply at that

**Review by j-wags — APPROVED** [review]
Awesome, thanks Matt!

**Comment by j-wags** [comment]
(There's a few things going on in this one and they'd all be ok to merge but I figure you have more context so I'll let you merge when you're ready)

**Comment by mattwthompson** [comment]
Thanks, the changes here should be inconsequential and/or limited to the low-stakes-and-often-broken-anyway beta workflow

**Pull request merged**